src/index.ts

```diff
@@ -232,28 +255,54 @@ class SOOSCSAAnalysis {
       );
 
       soosLogger.logLineSeparator();
-      soosLogger.info("Starting analysis scan");
-      await soosAnalysisApiClient.startScan({
+
+      await analysisService.startScan({
         clientId: this.args.clientId,
         projectHash,
         analysisId: analysisId,
+        scanType: ScanType.CSA,
+        scanUrl: result.scanUrl,
       });
 
-      soosLogger.info(
-        `Analysis scan started successfully, to see the results visit: ${result.scanUrl}`
-      );
+      const scanStatus = await analysisService.waitForScanToFinish({
+        scanStatusUrl: result.scanStatusUrl,
+        scanUrl: result.scanUrl,
+      });
 
-      if (this.args.onFailure === OnFailure.Fail) {
-        soosLogger.info("Waiting for scan to finish...");
-        await this.waitForScanToFinish({
-          apiClient: soosAnalysisApiClient,
-          scanStatusUrl: result.scanStatusUrl,
-          attempt: 0,
+      if (this.args.outputFormat !== undefined) {
+        await analysisService.generateFormattedOutput({
+          clientId: this.args.clientId,
+          projectHash: result.projectHash,
+          projectName: this.args.projectName,
+          branchHash: result.branchHash,
+          scanType: ScanType.CSA,
+          analysisId: result.analysisId,
+          outputFormat: this.args.outputFormat,
+          sourceCodePath: this.args.checkoutDir,
+          workingDirectory: this.args.checkoutDir,
         });
       }
+
+      if (this.args.onFailure === OnFailure.Fail) {
```

Comment:
we should report the status regardless, but then use this arg for the fail message and process exit

Reply:
seems like something we could have in the api client (not the process exit, but a "const reportStatus = (scanStatus: ScanStatusEnum): boolean" type method that returns whether we should exit or not